Add test dry run mode release notes (#25702)
gradle/gradle: subprojects/docs/src/docs/release/notes.md

See the [task options](userguide/custom_tasks.html#sec:listing_task_options) user manual section for more information.

### Dry run mode for test execution

It is possible now to run a test without actual test execution. 

> This can be useful for quick verifying of tests filtering configuration.

Reviewer: ```suggestion
This can be useful for quickly verifying which tests will actually run when [filtering a test suite](userguide/java_testing.html#test_filtering).
```

> Dry run mode can be enabled either by `--test-dry-run` command-line option or via [dryRun](dsl/org.gradle.api.tasks.testing.Test.html#org.gradle.api.tasks.testing.Test:dryRun) property.

Reviewer: ```suggestion
Dry run mode can be enabled by either the `--test-dry-run` command-line option or via the [dryRun](dsl/org.gradle.api.tasks.testing.Test.html#org.gradle.api.tasks.testing.Test:dryRun) property.
```

> This mode is compatible with JUnit and TestNG frameworks.

Reviewer: ```suggestion
This mode is compatible with the JUnit and TestNG frameworks.
```
